core/project_settings.h

```diff
@@ -218,4 +226,29 @@ Variant _GLOBAL_DEF_ALIAS(const String &p_var, const String &p_old_name, const V
 #define GLOBAL_DEF_ALIAS_RST(m_var, m_old_name, m_value) _GLOBAL_DEF(m_var, m_old_name, m_value, true)
 #define GLOBAL_GET(m_var) ProjectSettings::get_singleton()->get(m_var)
 
+/////////////////////////////////////////////////////////////////////////////////////////
+// Cached versions of GLOBAL_GET.
+// Cached but uses a typed variable for storage, this can be more efficient.
+// DO NOT USE Variants for this, as their lifetime cannot be static.
+// Only use for basic types, preferably through the predefined versions below.
+#define _GLOBAL_GET_CACHED(VAR_TYPE, SETTING_NAME) ([](const StringName &p_name) -> const VAR_TYPE & {\
+	static VAR_TYPE local_var;                                                   \
+		static uint32_t local_version = 0;                                      \
+		uint32_t new_version = ProjectSettings::get_singleton()->get_version(); \
+		if (local_version != new_version) {                                     \
+			local_version = new_version;                                        \
+			local_var = ProjectSettings::get_singleton()->get(p_name);     \
+		}					                                                   \
+		return local_var; })(SETTING_NAME)
```

Comment:
Any reason why the setting name needs to be passed in through the lambda call, thus paying the cost for the `StringName` construction every time, when you're only using it if the settings have changed?

Reply:
If it's non-literal string arg, it would need to be captured, which would fail here.
There aren't that many cases of this though, so it should probably be optimized as you suggest!

Reply:
I would just add a `[&]` capture on the lambda, along with inlining the setting name usage. I would be surprised if the generated code was any different for the string literal case, and storing a single reference for the non-literal case can't be that bad.